### DISCUSSION TITLE
How do we get dotenv working?

### DISCUSSION THREAD

**Discussion by hankwallace** [hubotio/hubot#1686] [Q&A]
In my previous bots, I had a script that was doing most of what dotenv does. I can't get the same to work with the new version of hubot and the dotenv docs have been no help.

**Comment by joeyguerra** [comment]
I currently use Node v20.6.0. It loads a `.env` file with the `--env-file=.env` CLI argument. Here's an example of my npm start scripts for prod and dev:

// package.json
```json
"start": "NODE_ENV=production node --env-file=.env ./node_modules/hubot/bin/hubot.js -a @hubot-friends/hubot-discord -n jbot",
"start:local": "NODE_ENV=development PORT=3001 node --env-file=.env --watch ./node_modules/hubot/bin/hubot.js -a @hubot-friends/hubot-discord --name jbot"
```

Prior to that I used the following:

// package.json
```json
"start": "env $(cat .env | grep -v \"#\" | xargs -0) hubot -a @hubot-friends/hubot-discord -n jbot"
```

I've never used the `dotenv` module. Are you willing to post what you did for your previous bots, perhaps a second pair of eyes might help?
